src/program.ts

```diff
@@ -2962,6 +2967,14 @@ export class File extends Element {
     if (!exports) this.exports = exports = new Map();
     exports.set(name, element);
     if (this.source.sourceKind == SourceKind.LIBRARY_ENTRY) this.program.ensureGlobal(name, element);
+
+    // Also, add to the namespaces that caputure our exports
+    for(let i = 0; i < this.importedNamespacesFromThisFile.length; i++) {
+      let ns = this.importedNamespacesFromThisFile[i];
+      if (!ns.lookup(name)) {
+        ns.add(name, element);
+      }
+    }
```

Comment:
```suggestion
    // Also add to all the `import * as X` alias namespaces
    let aliasNamespaces = this.importedNamespacesFromThisFile;
    for(let i = 0, k = aliasNamespaces .length; i < k; i++) {
      aliasNamespaces[i].add(name, element);
    }
```

Reply:
Just came up with the `aliasNamespaces` term while looking at this section, perhaps that name is a little more convenient than `importedNamespacesFromThisFile` in general? :)